### ISSUE TITLE
Any input produce the same output

### ISSUE THREAD

**Issue by Benedetta89** [BVLC/caffe#3842]
I've trained an AlexNet on my own data for a two-output classification problem. When i try to test it for each image in input i obtain the same value of probability in output. How doesn'it work??
these are the scripts i used:
THE NET
name: "AlexNet"
layer {
  name: "data"
  type: "Data"
  top: "data"
  transform_param {
    mirror: true
    mean_file: "/home/deepcuda/Scrivania/caffe/caffe/data/ilsvrc12/imagenet_mean_65_completo_new_label.binaryproto" 
  }
  data_param {
    source: "/home/deepcuda/Scrivania/caffe/caffe/dati_caffe/image_train_lmdb_65_new_not_skeleton"
    batch_size: 64
    backend: LMDB
  }
}
layer {
  name: "conv1"
  type: "Convolution"
  bottom: "data"
  top: "conv1"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 96
    kernel_size: 11
    stride: 1
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layer {
  name: "relu1"
  type: "ReLU"
  bottom: "conv1"
  top: "conv1"
}
layer {
  name: "norm1"
  type: "LRN"
  bottom: "conv1"
  top: "norm1"
  lrn_param {
    local_size: 5
    alpha: 0.0001
    beta: 0.75
  }
}
layer {
  name: "pool1"
  type: "Pooling"
  bottom: "norm1"
  top: "pool1"
  pooling_param {
    pool: MAX
    kernel_size: 3
    stride: 2
  }
}
layer {
  name: "conv2"
  type: "Convolution"
  bottom: "pool1"
  top: "conv2"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 256
    pad: 2
    kernel_size: 5
    group: 2
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0.1
    }
  }
}
layer {
  name: "relu2"
  type: "ReLU"
  bottom: "conv2"
  top: "conv2"
}
layer {
  name: "norm2"
  type: "LRN"
  bottom: "conv2"
  top: "norm2"
  lrn_param {
    local_size: 5
    alpha: 0.0001
    beta: 0.75
  }
}
layer {
  name: "pool2"
  type: "Pooling"
  bottom: "norm2"
  top: "pool2"
  pooling_param {
    pool: MAX
    kernel_size: 3
    stride: 2
  }
}
layer {
  name: "conv3"
  type: "Convolution"
  bottom: "pool2"
  top: "conv3"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 384
    pad: 1
    kernel_size: 3
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layer {
  name: "relu3"
  type: "ReLU"
  bottom: "conv3"
  top: "conv3"
}
layer {
  name: "conv4"
  type: "Convolution"
  bottom: "conv3"
  top: "conv4"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 384
    pad: 1
    kernel_size: 3
    group: 2
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0.1
    }
  }
}
layer {
  name: "relu4"
  type: "ReLU"
  bottom: "conv4"
  top: "conv4"
}
layer {
  name: "conv5"
  type: "Convolution"
  bottom: "conv4"
  top: "conv5"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 256
    pad: 1
    kernel_size: 3
    group: 2
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0.1
    }
  }
}
layer {
  name: "relu5"
  type: "ReLU"
  bottom: "conv5"
  top: "conv5"
}
layer {
  name: "pool5"
  type: "Pooling"
  bottom: "conv5"
  top: "pool5"
  pooling_param {
    pool: MAX
    kernel_size: 3
    stride: 2
  }
}
layer {
  name: "fc6"
  type: "InnerProduct"
  bottom: "pool5"
  top: "fc6"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 1024
    weight_filler {
      type: "gaussian"
      std: 0.005
    }
    bias_filler {
      type: "constant"
      value: 0.1
    }
  }
}
layer {
  name: "relu6"
  type: "ReLU"
  bottom: "fc6"
  top: "fc6"
}
layer {
  name: "drop6"
  type: "Dropout"
  bottom: "fc6"
  top: "fc6"
  dropout_param {
    dropout_ratio: 0.5
  }
}
layer {
  name: "fc7"
  type: "InnerProduct"
  bottom: "fc6"
  top: "fc7"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 512
    weight_filler {
      type: "gaussian"
      std: 0.005
    }
    bias_filler {
      type: "constant"
      value: 0.1
    }
  }
}
layer {
  name: "relu7"
  type: "ReLU"
  bottom: "fc7"
  top: "fc7"
}
layer {
  name: "drop7"
  type: "Dropout"
  bottom: "fc7"
  top: "fc7"
  dropout_param {
    dropout_ratio: 0.5
  }
}
layer {
  name: "fc8"
  type: "InnerProduct"
  bottom: "fc7"
  top: "fc8"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 2
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layer {
  name: "loss"
  type: "Softmax"
  bottom: "fc8"
  top: "loss"
}
THE SOLVER
# The train/test net protocol buffer definition
net: "/home/deepcuda/Scrivania/caffe/caffe/codice_retina/riproduzione_esempio/prova3_new.prototxt"
# The base learning rate, momentum and the weight decay of the network.
base_lr: 0.01
momentum: 0.9
weight_decay: 0.0005
# The learning rate policy
#- inv: return base_lr * (1 + gamma * iter) ^ (- power)
lr_policy: "inv"
gamma: 0.0001
power: 0.75
# Display every 100 iterations
display: 100
# The maximum number of iterations
# numero di iterazioni per la fase di training (quanti batch)
max_iter: 120000
# snapshot intermediate results
snapshot: 10000
snapshot_prefix: "/home/deepcuda/Scrivania/caffe/caffe/codice_retina/riproduzione_esempio/snapshot_prova3_new_not_skeleton/"
# solver mode: CPU or GPU
solver_mode: GPU
device_id:1
 
TEST:python script
import scipy
import numpy
from scipy import misc
import caffe
import sys
caffe_root = '/home/deepcuda/Scrivania/caffe/caffe/'  
sys.path.insert(0, caffe_root + 'python')
 

####################################caricamento pathname################################
pathdir="/home/deepcuda/Scrivania/caffe/caffe/codice_retina/riproduzione_esempio/output/"
pathdir1="/home/deepcuda/Scrivania/dataset/DRIVE.test.nlmeans/"
pathdir2="/home/deepcuda/Scrivania/dataset/test/manual1/"
pathdir3="/home/deepcuda/Scrivania/dataset/test/mask/"
test_imgName=[]
test_maskName=[]
test_grtName=[]
for num in range(1,21): 
    #string1=str(num)
    #string="_image.png"
    #string=pathdir1+string1+string2     
    #test_imgName.append(string)
    test_imgName.append(pathdir1+str(num)+"_image.png")
    test_grtName.append(pathdir2+str(num)+"_manual1.jpg")
    test_maskName.append(pathdir3+str(num)+"_test_mask.jpg")
    print(test_imgName[num-1])
    print(test_grtName[num-1])
    print(test_maskName[num-1])

test_original_images=numpy.zeros((20,584,565))
test_grt_images=numpy.zeros((20,584,565))
test_mask_images=numpy.zeros((20,584,565))
for num in range(1,21):
    test_original_images[num-1]=misc.imread(test_imgName[num-1])
    test_grt_images[num-1]=misc.imread(test_grtName[num-1],True)
    test_mask_images[num-1]=misc.imread(test_maskName[num-1],True)
print(test_original_images.shape)
caffe.set_mode_gpu()

model_def = caffe_root + 'codice_retina/riproduzione_esempio/prova4_new.prototxt'
model_weights = caffe_root + 'codice_retina/riproduzione_esempio/snapshot_prova4_new_not_skeleton/_iter_10000.caffemodel'

net = caffe.Net(model_def,      # defines the structure of the model
                model_weights,  # contains the trained weights
                caffe.TEST)     # use test mode (e.g., don't perform dropout)         
net.blobs['data'].reshape(1,        # batch size
                          1,         # 1-channel (BGR) images
            mu = numpy.load(caffe_root + 'data/ilsvrc12/imagenet_mean_65_new_label.npy')

transformer = caffe.io.Transformer({'data': net.blobs['data'].data.shape})

mu = mu.mean(1).mean(1)  # average over pixels to obtain the mean (BGR) pixel values
          
b=test_original_images[0:1,0:65,0:65]
print(b.shape)               
b=numpy.zeros((1,65,65))

transformed_image = transformer.preprocess('data',b) 
 
                         
net.blobs['data'].data[...] = b 
output=net.forward()
output_prob = output['loss'][0]
print(output_prob)

print(test_original_images[0].shape)

 
Any image i put into the network to test it 
THE OUTPUT PROBABILITY IS ALWAYS
[ 0.49922553  0.5007745 ]
WHY?????

**Comment by seanbell** [comment]
You should look at all of the activations in the network (not just the last layer).  One possibility is that the weights collapsed during training, so that the prediction only depends on the bias.  Another possibility is that you're not extracting the probabilities correctly, or you're not transferring the weights properly from training to test, or you're not using the correct intensity scale (255 vs 1) for input images.

I'm closing this since it's a question about how to use Caffe.  From https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md:
> _Please do not post usage, installation, or modeling questions, or other requests for help to Issues._
Use the [caffe-users list](https://groups.google.com/forum/#!forum/caffe-users) instead. This helps developers maintain a clear, uncluttered, and efficient view of the state of Caffe.

If you think you found a bug in caffe, please re-open the issue.  Otherwise, please ask future questions like this on the mailing list.
